trino/trino_test.go

```diff
@@ -686,6 +688,48 @@ func TestQueryFailure(t *testing.T) {
 	assert.IsTypef(t, new(ErrQueryFailed), err, "unexpected error: %w", err)
 }
 
+// This test ensures that the fetch method is not generating stack overflow errors.
+// === RUN   TestFetchNoStackOverflow
+// runtime: goroutine stack exceeds 1000000000-byte limit
+// runtime: sp=0x14037b00390 stack=[0x14037b00000, 0x14057b00000]
+// fatal error: stack overflow
+func TestFetchNoStackOverflow(t *testing.T) {
+	debug.SetMaxStack(50 * 1024)
```

Comment:
don't you need to reset it at the end of the method.
Also what if multiple tests are running at the same time.

Sorry for noob questions :)

Reply:
Those are valid questions. I ran all the tests with this one as the first and everything works. I assume all other tests don't require a bigger stack. We can either:
* revert this at the end of the test
* set it in `TestMain`

Tests don't run in parallel by default and it requires an explicit flag `-parallel`. I tried it now and some integration tests fail when checking for query status. I assume they're not written to support parallel execution, that is they don't use unique identifiers.

Reply:
If tests are not built to run in parallel then I would revert at the end of test via `defer`.

Reply:
That's done already :)

```
previousSetting := debug.SetMaxStack(50 * 1024)
defer debug.SetMaxStack(previousSetting)
```

Note that i also had to increase the timeout of the tests on the CI.